bittensor_cli/src/commands/stake/children_hotkeys.py

```diff
@@ -0,0 +1,711 @@
+
+                hotkey = Text(hotkey, style="italic red" if proportion == 0 else "")
+                table.add_row(
+                    str(netuid),
+                    hotkey,
+                    proportion_str,
+                    take_str,
+                    str(f"{stake_weight:.3f}"),
+                )
+
+            avg_take_per_netuid = avg_take_per_netuid / len(children_info)
+
+            # add totals row for this netuid
+            table.add_row(
+                "",
+                "[dim]Total[/dim]",
+                f"[dim]{total_proportion_per_netuid:.3f}%[/dim]",
+                f"[dim](avg) {avg_take_per_netuid * 100:.3f}%[/dim]",
+                f"[dim]{total_stake_weight_per_netuid:.3f}τ[/dim]",
+                style="dim",
+            )
+
+            # add to grand totals
+            total_proportion += total_proportion_per_netuid
+            total_stake_weight += total_stake_weight_per_netuid
+
+            # Add a dividing line if there are more than one netuid
+            if len(netuid_children_tuples) > 1:
+                table.add_section()
+
+        console.print(table)
+
+    # Core logic for get_children
+    if netuid is None:
+        # get all netuids
+        netuids = await subtensor.get_all_subnet_netuids()
+        await get_total_stake_for_hk(wallet.hotkey.ss58_address, True)
+        netuid_children_tuples = []
+        for netuid in netuids:
+            success, children, err_mg = await subtensor.get_children(
+                wallet.hotkey.ss58_address, netuid
+            )
+            if children:
+                netuid_children_tuples.append((netuid, children))
+            if not success:
+                err_console.print(
+                    f"Failed to get children from subtensor {netuid}: {err_mg}"
+                )
+        await _render_table(wallet.hotkey.ss58_address, netuid_children_tuples)
+    else:
+        success, children, err_mg = await subtensor.get_children(
+            wallet.hotkey.ss58_address, netuid
+        )
+        if not success:
+            err_console.print(f"Failed to get children from subtensor: {err_mg}")
+        await get_total_stake_for_hk(wallet.hotkey.ss58_address, True)
+        if children:
+            netuid_children_tuples = [(netuid, children)]
+            await _render_table(wallet.hotkey.ss58_address, netuid_children_tuples)
+
+        return children
+
+
+async def set_children(
+    wallet: Wallet,
+    subtensor: "SubtensorInterface",
+    children: list[str],
+    proportions: list[float],
+    netuid: Optional[int] = None,
+    wait_for_inclusion: bool = True,
+    wait_for_finalization: bool = True,
+):
+    """Set children hotkeys."""
+    # Validate children SS58 addresses
+    for child in children:
+        if not is_valid_ss58_address(child):
+            err_console.print(f":cross_mark:[red] Invalid SS58 address: {child}[/red]")
+            return
+
+    total_proposed = sum(proportions)
+    if total_proposed > 1:
```

Comment:
I really think we should consider decimals. Probably not for this iteration, because we need to ship this, but probably for the next improvement.